packages/next/src/client/components/router-reducer/reducers/server-action-reducer.ts

```diff
@@ -0,0 +1,155 @@
+import {
+  ActionFlightData,
+  FlightData,
+} from '../../../../server/app-render/types'
+import { callServer } from '../../../app-call-server'
+import {
+  NEXT_ROUTER_STATE_TREE,
+  NEXT_URL,
+  RSC_CONTENT_TYPE_HEADER,
+} from '../../app-router-headers'
+import { createRecordFromThenable } from '../create-record-from-thenable'
+import { readRecordValue } from '../read-record-value'
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { createFromFetch } from 'react-server-dom-webpack/client'
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { encodeReply } from 'react-server-dom-webpack/client'
+
+import {
+  PrefetchKind,
+  ReadonlyReducerState,
+  ReducerState,
+  ServerActionAction,
+} from '../router-reducer-types'
+import { addBasePath } from '../../../add-base-path'
+import { createHrefFromUrl } from '../create-href-from-url'
+import { getRedirectError } from '../../redirect'
+
+type FetchServerActionResult = {
+  result: ActionFlightData | undefined
+  redirectLocation: URL | undefined
+}
+
+async function fetchServerAction(
+  state: ReadonlyReducerState,
+  { actionId, actionArgs }: ServerActionAction
+): Promise<FetchServerActionResult> {
+  const body = await encodeReply(actionArgs)
+
+  const res = await fetch('', {
+    method: 'POST',
+    headers: {
+      Accept: RSC_CONTENT_TYPE_HEADER,
+      'Next-Action': actionId,
+      [NEXT_ROUTER_STATE_TREE]: JSON.stringify(state.tree),
+      ...(state.nextUrl
+        ? {
+            [NEXT_URL]: state.nextUrl,
+          }
+        : {}),
+    },
+    body,
+  })
+
+  if (!res.ok) {
+    throw new Error(await res.text())
+  }
```

Comment:
Does this resume the digest or that will be handled by Flight (`await createFromFetch(Promise.resolve(res))`)?

Reply:
as discussed on slack, we never actually send down any errors so there's no digest.